lib/ddg-atbgrabber.js

```diff
@@ -25,11 +26,15 @@ exports.install = function () {
       contentScriptFile: './js/duckduckgo.js',
       attachTo: ["existing", "top"],
       onAttach: function(worker) {
-        worker.port.on('loaded', function(atbParam) {
+        worker.port.on('loaded', function(atbParam, hideToolbarIcon) {
           // set the ATB param from the loaded site if it was not set yet
           if (atbParam && !ss.storage.atb_set) {
             ss.storage.atb = atbParam;
             ss.storage.atb_set = atbParam;
+
+            if (hideToolbarIcon) {
+              prefSet.prefs['toolbar_button'] = false;
```

Comment:
This doesn't work. I think this code is running before the `onPerfChange` callback has been registered here https://github.com/duckduckgo/firefox-zeroclickinfo/blob/master/lib/main.js#L48

You could try this just to be safe.
```
prefSet.prefs['toolbar_button'] = false;
 ui.uninstallToolbarButton();
```